tools/integration_tests/emulator_tests/util/test_helper.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package emulator_tests
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/setup"
+)
+
+// StartProxyServer starts a proxy server as a background process and handles its lifecycle.
+//
+// It launches the proxy server with the specified configuration and port, logs its output to a file,
+// and sets up signal handling for graceful shutdown.
+//
+// The function also starts a simple HTTP server to keep the process running and provides an
+// endpoint to check the proxy status.
+//
+// When receiving SIGINT or SIGTERM signals, it gracefully shuts down the proxy server by:
+//   - Sending SIGINT to the proxy process.
+//   - Killing any processes listening on given port.
+func StartProxyServer(port int, configPath string) {
+	// Start the proxy in the background
+	cmd := exec.Command("go", "run", "../proxy_server/.", "--config-path="+configPath)
```

Comment:
This requires go in the path right. Do we have other way?

Reply:
This is with respect to test package directory. So the path will be always same.
e.g.
```
emulator_tests/write_stall
emulator_tests/proxy_server/
```